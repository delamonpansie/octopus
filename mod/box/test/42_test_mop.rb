#!/usr/bin/ruby
# encoding: ASCII

$: << File.dirname($0) + '/lib'
boxpath = File.realpath(File.dirname(File.dirname($0)))
require 'run_env'

class Env < RunEnv
  def config
    super + <<EOD
object_space[0].enabled = 1
object_space[0].index[0].type = "HASH"
object_space[0].index[0].unique = 1
object_space[0].index[0].key_field[0].fieldno = 0
object_space[0].index[0].key_field[0].type = "NUM"

EOD
  end
end

env = Env.new
env.cd do
  ln_s '../mop.ml', '.'
  ln_s File.join(boxpath, 'src-ml/box1.cmi'), 'box1.cmi'
  `ocamlopt.opt -O3 -g -annot -I . -I +../batteries -shared -ccopt "-Wl,-Bsymbolic -Wl,-z,now -Wl,-z,combreloc" mop.ml -o mop_1.cmxs`
end

env.connect_eval do
  5.times {|i| insert [i, i.to_s] }
  lua "user_proc.mop"
  env.stop
  puts `./octopus --cat 00000000000000000002.xlog | sed 's/tm:[^ ]* //'` + "\n"
end


env.connect_eval do
  select 1,2
end
