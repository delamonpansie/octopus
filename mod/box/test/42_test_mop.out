# box.insert([0, "0"])
1

# box.insert([1, "1"])
1

# box.insert([2, "2"])
1

# box.insert([3, "3"])
1

# box.insert([4, "4"])
1

# box.lua("user_proc.mop")
[]

lsn:2 shard:0 scn:2 t:wal/usr13 INSERT n:0 flags:00000000 <0:"\x00\x00\x00\x00", "0">
lsn:3 shard:0 scn:3 t:wal/usr13 INSERT n:0 flags:00000000 <1:"\x01\x00\x00\x00", "1">
lsn:4 shard:0 scn:4 t:wal/usr13 INSERT n:0 flags:00000000 <2:"\x02\x00\x00\x00", "2">
lsn:5 shard:0 scn:5 t:wal/usr13 INSERT n:0 flags:00000000 <3:"\x03\x00\x00\x00", "3">
lsn:6 shard:0 scn:6 t:wal/usr13 INSERT n:0 flags:00000000 <4:"\x04\x00\x00\x00", "4">
lsn:7 shard:0 scn:7 t:wal/tlv BOX_MULTY { UPDATE_FIELDS n:0 flags:00000001 <1:"\x01\x00\x00\x00"> [field_no:1 op:set <"1">] ; UPDATE_FIELDS n:0 flags:00000001 <1:"\x01\x00\x00\x00"> [field_no:1 op:set <12850:"22">] ; UPDATE_FIELDS n:0 flags:00000001 <1:"\x01\x00\x00\x00"> [field_no:1 op:set <13107:"33">] ; UPDATE_FIELDS n:0 flags:00000001 <1:"\x01\x00\x00\x00"> [field_no:1 op:set <13364:"44">] ; DELETE n:0 flags:00000001 <2:"\x02\x00\x00\x00">; INSERT n:0 flags:00000003 <2:"\x02\x00\x00\x00", "42!">;  }

# box.select(1, 2)
[["\x01\x00\x00\x00", "44"], ["\x02\x00\x00\x00", "42!"]]

